packages/hierarchies-react/src/presentation-hierarchies-react/UnifiedSelectionContext.tsx

```diff
@@ -0,0 +1,62 @@
+import { createContext, PropsWithChildren, useContext, useEffect, useState } from "react";
+import { Key } from "@itwin/presentation-common";
+
+interface UnifiedSelectionContext {
+  has: (keys: Key[]) => boolean;
+  add: (keys: Key[]) => void;
+  remove: (keys: Key[]) => void;
+}
+
+const unifiedSelectionContext = createContext<UnifiedSelectionContext>({
+  has: () => false,
+  add: () => {},
+  remove: () => {},
+});
+
+/** @beta */
+export interface EventLike<TArgs> {
+  addListener: (listener: (args: TArgs) => void) => () => void;
+}
+
+/** @beta */
+export interface UnifiedSelectionContainer {
+  add: (keys: Key[]) => void;
+  remove: (keys: Key[]) => void;
+  has: (keys: Key[]) => boolean;
+}
+
+/** @beta */
+export interface UnifiedSelectionStore {
+  onChange: EventLike<UnifiedSelectionContainer>;
+  container: UnifiedSelectionContainer;
+}
+
+interface Props {
+  store: UnifiedSelectionStore;
+}
+
+/** @beta */
+export function UnifiedSelectionContextProvider({ store, children }: PropsWithChildren<Props>) {
+  const [value, setValue] = useState<UnifiedSelectionContext>({
+    has: (keys) => store.container.has(keys),
+    add: (keys) => store.container.add(keys),
```

Comment:
I know integration with the new `@itwin/unified-selection` is still a TODO, but just noting - we'll want to support "replace", too.

Reply:
Yes. I think I will need to reiterate unified selection handling in tree when integrating the new package.